src/nvim/api/buffer.c

```diff
@@ -636,6 +634,297 @@ ArrayOf(Integer, 2) buffer_get_mark(Buffer buffer, String name, Error *err)
   return rv;
 }
 
+/// Returns mark info at the given position or mark index
+///
+/// @param buffer The buffer handle
+/// @param namespace a identifier returned previously with namespace_create
+/// @param id [row, col] or mark_id of mark
+/// @param[out] err Details of an error that may have occurred
+/// @return [mark_id, row, col]
+ArrayOf(Object) buffer_mark_index(Buffer buffer, Integer namespace, Object id, Error *err)
```

Comment:
Hmm, surely you mean _after_ (or in)  #4934. We don't want to have both `vim_old_name`, `vim_new_name` and `nvim_new_name`, only the first and the last.

But then again, I don't see why we are forced to mass rename existing API names (if that was what you meant) in order for extmark ( one specific feature that happens to have many API functions) to have have a "sub-"namespace. We can just do the latter if we deem it warranted.